stdlib/public/core/Arrays.swift.gyb

```diff
@@ -503,6 +503,86 @@ public struct ${Self}<Element>
+    // index that is required for Array performance.  The optimizer is not
+    // capable of creating partial specializations yet.
+    // NOTE: Range checks are not performed here, because it is done later by
+    // the subscript function.
+    return i + 1
+  }
+
+  public func formIndex(after i: inout Int) {
+    // NOTE: this is a manual specialization of index movement for a Strideable
+    // index that is required for Array performance.  The optimizer is not
+    // capable of creating partial specializations yet.
+    // NOTE: Range checks are not performed here, because it is done later by
+    // the subscript function.
+    i += 1
+  }
+
+  @warn_unused_result
+  public func index(before i: Int) -> Int {
+    // NOTE: this is a manual specialization of index movement for a Strideable
+    // index that is required for Array performance.  The optimizer is not
+    // capable of creating partial specializations yet.
+    // NOTE: Range checks are not performed here, because it is done later by
+    // the subscript function.
+    return i - 1
+  }
+
+  public func formIndex(before i: inout Int) {
+    // NOTE: this is a manual specialization of index movement for a Strideable
+    // index that is required for Array performance.  The optimizer is not
+    // capable of creating partial specializations yet.
+    // NOTE: Range checks are not performed here, because it is done later by
+    // the subscript function.
+    i -= 1
+  }
+
+  @warn_unused_result
+  public func index(_ i: Int, offsetBy n: Int) -> Int {
+    // NOTE: this is a manual specialization of index movement for a Strideable
+    // index that is required for Array performance.  The optimizer is not
+    // capable of creating partial specializations yet.
+    // NOTE: Range checks are not performed here, because it is done later by
+    // the subscript function.
+    return i + n
+  }
+
+  @warn_unused_result
+  public func index(
+    _ i: Int, offsetBy n: Int, limitedBy limit: Int
+  ) -> Int? {
+    // NOTE: this is a manual specialization of index movement for a Strideable
+    // index that is required for Array performance.  The optimizer is not
+    // capable of creating partial specializations yet.
+    // NOTE: Range checks are not performed here, because it is done later by
+    // the subscript function.
+    let l = limit - i
+    if n > 0 ? l >= 0 && l < n : l <= 0 && n < l {
+      return nil
+    }
+    return i + n
+  }
+
+  @warn_unused_result
+  public func distance(from start: Int, to end: Int) -> Int {
+    // NOTE: this is a manual specialization of index movement for a Strideable
+    // index that is required for Array performance.  The optimizer is not
+    // capable of creating partial specializations yet.
+    // NOTE: Range checks are not performed here, because it is done later by
+    // the subscript function.
+    return end - start
+  }
+
+  public func _failEarlyRangeCheck(_ index: Int, bounds: Range<Int>) {
+  }
```

Comment:
Could you add a comment that says that this method is a no-op on purpose for performance reasons?